File: rag-backend/src/main/java/cn/sdh/backend/entity/Announcement.java
package cn.sdh.backend.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("announcement")
public class Announcement {

    @TableId(type = IdType.AUTO)
    private Long id;

    private String title;

    private String content;

    private String type;

    private Integer priority;

    private String targetType;

    private String targetIds;

    private LocalDateTime publishTime;

    private LocalDateTime expireTime;

    private Integer isTop;

    private Long publisherId;

    private Integer status;

    private LocalDateTime createTime;

    private LocalDateTime updateTime;
}

File: rag-frontend/src/api/announcement.ts
import request from '@/utils/request'
import type { ApiResponse, PageResult } from '@/types'

export interface Announcement {
  id: number
  title: string
  content: string
  type: string
  priority: number
  targetType: string
  targetIds?: string
  publishTime?: string
  expireTime?: string
  isTop: number
  publisherId?: number
  status: number
  createTime?: string
  updateTime?: string
}

export interface AnnouncementDetail {
  announcement: Announcement
  readCount: number
  isRead?: boolean
}

export function getAnnouncementList(params: {
  page: number
  pageSize: number
  type?: string
  status?: number
}): Promise<ApiResponse<PageResult<Announcement>>> {
  return request.get('/api/announcement/list', { params })
}

export function getActiveAnnouncements(): Promise<ApiResponse<Announcement[]>> {
  return request.get('/api/announcement/active')
}

export function getAnnouncementDetail(id: number): Promise<ApiResponse<AnnouncementDetail>> {
  return request.get(`/api/announcement/${id}`)
}

export function createAnnouncement(data: Partial<Announcement>): Promise<ApiResponse<null>> {
  return request.post('/api/announcement', data)
}

export function updateAnnouncement(id: number, data: Partial<Announcement>): Promise<ApiResponse<null>> {
  return request.put(`/api/announcement/${id}`, data)
}

export function publishAnnouncement(id: number): Promise<ApiResponse<null>> {
  return request.put(`/api/announcement/${id}/publish`)
}

export function deleteAnnouncement(id: number): Promise<ApiResponse<null>> {
  return request.delete(`/api/announcement/${id}`)
}

export function markAnnouncementRead(id: number): Promise<ApiResponse<null>> {
  return request.post(`/api/announcement/${id}/read`)
}